### PR TITLE
provider/core: Created an endpoint for getting a specific snapshot version from buckets

### PR DESCRIPTION
**Pull request by nwwebb**
Added an endpoint for getting a snapshot by timestamp. Did I add the correct authentication annotation @ttomsu? PTAL @lwander @duftler

### CODE REVIEW

File: front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/SnapshotsController.groovy [hunk 1]
```diff
@@ -57,6 +57,20 @@ class SnapshotsController {
         return snapshotDAO.findById(id)
     }
 
+    @PostAuthorize("hasPermission(returnObject.application, 'APPLICATION', 'READ')")
+    @RequestMapping(value = "/{id:.+}/{timestamp:.+}", method = RequestMethod.GET)
+    Snapshot getVersionByTimestamp(@PathVariable String id,
+                        @PathVariable String timestamp,
```
Comment (lwander) [inline front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/SnapshotsController.groovy]: Indentation is off here

File: front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/SnapshotsController.groovy [hunk 2]
```diff
@@ -57,6 +57,20 @@ class SnapshotsController {
...
+                        @RequestParam(value = "limit", defaultValue = "20") int limit) {
+        def creationTime = timestamp.toLong()
+        for (Snapshot snapshot: snapshotDAO.getHistory(id, limit)) {
```
Comment (lwander) [inline front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/SnapshotsController.groovy]: This should probably be
` snapshotDAO.getHistory(id, limit)?.find { snapshot -> snapshot.timestamp == creationTime }`

**Comment by nwwebb** [comment]
@lwander addressed comments

**Comment by lwander** [comment]
LGTM, any concerns, @duftler?

**Comment by duftler** [comment]
LGTM.

**Pull request merged**